python/cudf/cudf/core/frame.py

```diff
@@ -1124,7 +1124,7 @@ def fillna(
                 for key, value in value.items()
             }
 
-        filled_data = {}
+        columns_list = []
```

Comment:
If we instead started with  a `ColumnAcccessor(multiindex=True)` here, would that have worked too?

Reply:
I think `ColumnAccessor._create_unsafe` is the one we need to use here.

Reply:
> I think `ColumnAccessor._create_unsafe` is the one we need to use here.

Updated code with this approach.